synapse/storage/databases/main/search.py

```diff
@@ -125,9 +125,6 @@ def __init__(
     ):
         super().__init__(database, db_conn, hs)
 
-        if not hs.config.server.enable_search:
-            return
```

Comment:
It isn't really clear to me what the ramifications of this are, does it mean that we'll now run all these updates when search is disabled? Or just that they're now registered and the code "exists" to handle those?

Reply:
The older updates will now run when upgrading from very old Synapse versions, instead of raising endless errors when search is disabled. Up-to-date Synapse installs will only try to run the delete_non_strings update.

I'm reasoning that if search is disabled, event_search is probably empty (unless search was enabled in the past) and the updates will complete quickly.